src/Leela.cpp

```diff
@@ -316,5 +316,6 @@ int main (int argc, char *argv[]) {
         }
     }
 
+    std::cout << std::endl;
```

Comment:
Maybe it should go in the "eof or other error" branch? That's the case you're hitting, right?

I checked GTP quit vs exit, and:
https://github.com/gcp/leela-zero/blob/47ddf38327b80ec5f2303344d1d9c65e25f5a3fb/src/GTP.cpp#L209

https://github.com/gcp/leela-zero/blob/47ddf38327b80ec5f2303344d1d9c65e25f5a3fb/src/GTP.cpp#L233

So they won't hit this case. quit is handled correctly and it needs to be according to: 
"The full response of this command must be sent before the                 engine closes the connection. The controller must receive                 the response before the connection is closed on its side."

Maybe "exit" should just be removed, OTOH it's handy if someone starts the binary and gets the typical 'how do I get out?!' reaction.

Reply:
I moved it under the "eof or other error" branch, which probably covers 90% of my problems (control-c being the others and that's harder to catch/fix).

this is a good compromise (fixes most of this problem and doesn't affect GTP at all)